src/util.cpp

```diff
@@ -221,30 +221,23 @@ boost::filesystem::path GetDebugLogPath()
     }
 }
 
-bool OpenDebugLog()
+std::string LogConfigFilter()
 {
-    boost::call_once(&DebugPrintInit, debugPrintInitFlag);
-    boost::mutex::scoped_lock scoped_lock(*mutexDebugLog);
+    // With no -debug flags, show errors and LogPrintf lines.
+    std::string filter = "error,main=info";
 
-    assert(fileout == NULL);
-    assert(vMsgsBeforeOpenLog);
-
-    boost::filesystem::path pathDebug = GetDebugLogPath();
-    fileout = fopen(pathDebug.string().c_str(), "a");
-    if (!fileout) {
-        return false;
-    }
-
-    setbuf(fileout, nullptr); // unbuffered
-    // dump buffered messages from before we opened the log
-    while (!vMsgsBeforeOpenLog->empty()) {
-        FileWriteStr(vMsgsBeforeOpenLog->front(), fileout);
-        vMsgsBeforeOpenLog->pop_front();
+    auto& categories = mapMultiArgs["-debug"];
+    std::set<std::string> setCategories(categories.begin(), categories.end());
+    if (setCategories.count(string("")) != 0 || setCategories.count(string("1")) != 0) {
+        // Turn on the firehose!
+        filter = "info";
```

Comment:
I'd like to make "debug" the firehose, so that "important" progress messages (that currently use `LogPrintf`) can be "info" events with a more specific category). But that can wait to a follow-up PR.

Reply:
+1 --- that tends to be the convention in most existing Rust projects that use `tracing`.

Reply:
Addressed in #4657.